File: insertion_sort/InsertionSort.java
import java.util.*;

/**
 * InsertionSort
 */
public class InsertionSort {

  public static void main(String[] args) {
    int inputArr[] = {7, 3, 10, 2, 1, 6, 5, 9, 8, 4};

    // variables
    int i = 1;
    int k = 0;
    int tmp = 0;

    int len = inputArr.length;

    long start = System.nanoTime();

    System.out.println("=== Exec Insertion Sort in Java ===");

    while(i < len) {
      tmp = inputArr[i];
      k = i;

      while((k > 0) && (inputArr[k - 1] > tmp)) {
        inputArr[k] = inputArr[k - 1];
        --k;
      }

      inputArr[k] = tmp;
      i++;
    }

    System.out.println(Arrays.toString(inputArr));

    long end = System.nanoTime();

    System.out.println("Exec time: " + (end - start) / 1000000f + " ms");
  }
}

File: insertion_sort/InsertionSort.rb
require 'time'

class InsertionSort
  input_arr = [7, 3, 10, 2, 1, 6, 5, 9, 8, 4]

  i = 1
  k = 0
  tmp = 0

  len = input_arr.size

  start_time = Time.now

  puts "=== Exec Insertion Sort in Ruby ==="

  while i < len
    tmp = input_arr[i]
    k = i

    while (k > 0) && (input_arr[k - 1] > tmp)
      input_arr[k] = input_arr[k - 1]
      k -= 1
    end

    input_arr[k] = tmp
    i += 1
  end

  puts input_arr.to_s

  end_time = Time.now

  puts "Exec time: #{(end_time - start_time) * 100000} ms"
end